src/plugins/intel_cpu/src/utils/cpu_utils.hpp

```diff
@@ -122,7 +122,9 @@ inline InferenceEngine::Precision normalizeToSupportedPrecision(InferenceEngine:
             break;
         }
         case InferenceEngine::Precision::FP16: {
+#if !defined(__ARM_FEATURE_FP16_VECTOR_ARITHMETIC)
             precision = InferenceEngine::Precision::FP32;
```

Comment:
Lets add "TODO: seems like can be enabled by default for all platforms (require performance benchmarking)".

Reply:
I think we should completely avoid using this 
#if defined(__ARM_FEATURE_FP16_VECTOR_ARITHMETIC)
except the logic related to the inference precision hint 
If we decided that inference precision will be fp16, all the code logic after this should not check for the availability of fp16 support.
Also If the model turned out to be fp16 and running in the environment without fp16 support, should we still handle this? By converting to fp32 or something.

Reply:
Agree. better to avoid the marco usage here. That's why I asked to add follow-up task on benchmarking, so the idea is to align behavior.
Anyway this function is mostly used to align precisions between infer_request and graph inputs/outputs. That logic should be removed in bounds of API 2.0 migration: all precision conversion will be handled as explicit Convert operations inside the graph (inserted by ConvertPrecision transformation).